test/wallet-rpc-test.js

```diff
@@ -0,0 +1,287 @@
+      assert.equal(response.address, receive);
+    });
+
+    // Select the watch only wallet that was created with the xpub
+    // at account index 0', derive the same address locally
+    // and pass to getaddressinfo, assert that ismine is true,
+    // select the standard wallet that has a different hd seed
+    // and then pass the same address to getaddressinfo and
+    // assert that ismine is false
+    it('should detect owned address', async () => {
+      // m/44'/1'/0'/0/0
+      const receive = 'mkpZhYtJu2r87Js3pDiWJDmPte2NRZ8bJV';
+      {
+        await wclient.execute('selectwallet', [watchOnlyWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.ismine, true);
+      }
+      {
+        await wclient.execute('selectwallet', [standardWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.ismine, false);
+      }
+    });
+
+    it('should detect a p2sh address', async () => {
+      const script = Script.fromMultisig(2, 2, pubkeys);
+
+      const address = Address.fromScript(script);
+      const addr = address.toString(network);
+      const response = await wclient.execute('getaddressinfo', [addr]);
+      assert.equal(response.isscript, true);
+      assert.equal(response.iswitness, false);
+      assert.equal(response.witness_program, undefined);
+    });
+
+    // assert that the correct witness program is returned for a
+    // p2wpkh address
+    it('should return the correct program', async () => {
+      // m/44'/1'/0'/0/5
+      const receive = 'bcrt1q53724q6cywuzsvq5e3nvdeuwrepu69jsc6ulmx';
+      const addr = Address.fromString(receive);
+
+      await wclient.execute('selectwallet', [watchOnlyWalletId]);
+      const str = addr.toString(network);
+      const response = await wclient.execute('getaddressinfo', [str]);
+      assert.equal(response.witness_program, addr.hash.toString('hex'));
+    });
+
+    // build a simple multisig script, create an address and send it to
+    // 'getaddressinfo', assert that it returns isscript as true and the
+    // correct value as the witness program
+    it('should detect p2wsh program', async () => {
+      const script = Script.fromMultisig(2, 2, pubkeys);
+      const address = Address.fromWitnessScripthash(script.sha256());
+      const addr = address.toString(network);
+      const response = await wclient.execute('getaddressinfo', [addr]);
+
+      assert.equal(response.isscript, true);
+      assert.equal(response.witness_program, address.hash.toString('hex'));
+    });
+
+    // Iterate through the lookahead amount of addresses and request
+    // the RPC 'getaddressinfo' for each, assert that each ismine is
+    // true and then assert that ismine is false when the next address
+    // generated is outside of the lookahead range
+    it('should detect ismine up to the lookahead', async ($) => {
```

Comment:
What is the `$`?

Reply:
Its a special character that can be used to skip tests, since `xit` doesn't exist in bmocha, you can use `$.skip()`

Reply:
`it.skip(` works fine. The `$` is not needed here right?